### ISSUE TITLE
LSP Support

### ISSUE THREAD

**Issue by akiyosi** [akiyosi/gonvim#19]
Gonvim supports neovim's build-in LSP.
This feature will be add in v 0.4.0.

**Comment by akiyosi** [comment]
This feature will be considered for addition in v0.5.0.
Because the addition on the neovim side is nvim v0.5.0

**Comment by pierreglaser** [comment]
Hi! Does the current version of `goneovim` supports `neovim`'s built-in LSP?

**Comment by akiyosi** [comment]
Hi :)
Goneovim is just a UI frontend for neovim currently. So, if you have build-in LSP enabled as neovim config (which would actually add the plugin), of course it can also be used with goneovim.

A related feature to the built-in LSP is the ability to display an icon based on the completion entry's kind when ExtPopupmenu is enabled.

<img width="690" alt="2020-07-16 23 21 13" src="https://user-images.githubusercontent.com/8478977/87683774-6de93300-c7bc-11ea-9fc7-38fe9169330b.png">

However, my personal opinion is that ExtPopupmenu isn't quite practical enough yet, and I'd recommend using Neovim's native completion menu.

**Comment by pierreglaser** [comment]
> However, my personal opinion is that ExtPopupmenu isn't quite practical enough yet, and I'd recommend using Neovim's native completion menu.

One thing that does not help using `ExtPopupmenu` is that LSP-complete items are displayed with an additional popup window (used to display the item docs) that sometimes conflicts (visually) with the `ExtPopupmenu`. Could the additional popup window be handled by `goneovim` as well?

Another thing I thought about when I mentionned `LSP support` was  to have the `Lint` section of the external statusline to integrate with the `LSP` diagnostics. Do you have any thoughts on this?

**Comment by akiyosi** [comment]
> One thing that does not help using ExtPopupmenu is that LSP-complete items are displayed with an additional popup window (used to display the item docs) that sometimes conflicts (visually) with the ExtPopupmenu. Could the additional popup window be handled by goneovim as well?

Yes, this is exactly the point at which I decide that we are not at the practical stage. The additional popup information is not available in [neovim ui](https://neovim.io/doc/user/ui.html).
And, it's not handled by goneovim. It's probably being output by the plugin, which is an LSP client. A mechanism needs to be provided to allow the plugin to control this additional popup information. I think.

> Another thing I thought about when I mentionned LSP support was to have the Lint section of the external statusline to integrate with the LSP diagnostics. Do you have any thoughts on this?

I don't have any solid ideas at this time.
Also, I'm not aggressive about improving the external statusline feature of goneovim. The reason is that neovim core users prefer to use own customized statusline.

On the other hand, neovim has an idea to externalize statusline, as well as other externalization mechanisms such as tab lines, popupmenu, and commandline.
I hope this concept will be realized.

> By the way, the editor overall has a great feel! thanks for working on this!

Thanks! :)
